src/arkode/arkode_mristep_io.c

```diff
@@ -431,7 +431,7 @@ int MRIStepSetOrder(void* arkode_mem, int ord)
   if (retval) { return (retval); }
 
   /* check for illegal inputs */
-  if (ord < 3 || ord > 4) { step_mem->q = 3; }
+  if (ord <= 0) { step_mem->q = 3; }
```

Comment:
What about `ord > 4`?

Reply:
This edit was done to be consistent with what's done for ARK and ERK:
https://github.com/LLNL/sundials/blob/2abd63bd6cbc354fb4861bba8e98d0b95d65e24a/src/arkode/arkode_arkstep_io.c#L1047
https://github.com/LLNL/sundials/blob/2abd63bd6cbc354fb4861bba8e98d0b95d65e24a/src/arkode/arkode_erkstep_io.c#L497

Across ARKODE, we check for illegal values, i.e. `ord > 4` during initialization. I would argue that we should fail early and raise an error in setters, not initialization. That is probably a more sweeping change that shouldn't be address in this PR, so I just made the behavior consistent across ARKODE.